sqlalchemy_redshift/dialect.py

```diff
@@ -641,23 +643,85 @@ def keyfunc(column):
             'redshift_interleaved_sortkey': interleaved_sortkey,
         }
 
+    def _do_generate_iam_auth_token(self, redshift, cparams):
+        # Hostname is the cluster identifier
+        cluster_id = cparams['host']
+        response = redshift.describe_clusters(ClusterIdentifier=cluster_id)
```

Comment:
Is there a way to have this be given instead of using a describeClusters call? At scale this is a great way to get rate limited.

Reply:
i would say to do something like:

```
hostname = cparams.get('hostname')
port = cparams.get('port')
if cparams.get('host'):
    response = redshift.describe_clusters(ClusterIdentifier=cluster_id)
    ...
    hostname = ...
    port = ...
```

or similar